core/src/main/resources/error/error-classes.json

```diff
@@ -225,14 +230,14 @@
           "The <functionName> should all be of type map, but it's <dataType>."
         ]
       },
-      "MAP_CONTAINS_KEY_DIFF_TYPES" : {
+      "MAP_FROM_ENTRIES_WRONG_TYPE" : {
```

Comment:
Can't we re-use the existing error sub-class `UNEXPECTED_INPUT_TYPE`?

Reply:
looks ok, let me re-use `UNEXPECTED_INPUT_TYPE`

Reply:
@MaxGekk 
Are you talking about `MAP_FROM_ENTRIES_WRONG_TYPE`? 
This is not a new subclass introduced by this pr, but a new one created by https://github.com/apache/spark/pull/38197/files. Let me file a new jira to replace `MAP_FROM_ENTRIES_WRONG_TYPE` with `UNEXPECTED_INPUT_TYPE`, because it is not related to this pr

Reply:
SPARK-41027, but if use `UNEXPECTED_INPUT_TYPE` , the `requiredType` should be `Array of pair Struct`

```
"requiredType" -> s"${toSQLType(ArrayType)} of pair ${toSQLType(StructType)}"
```

is it ok?

Reply:
https://github.com/apache/spark/pull/38507